attributes/default.rb

```diff
@@ -88,9 +93,16 @@
 # Set prefix to '' if you want Chef tags to be sent without prefix.
 default['datadog']['tag_prefix'] = 'tag:'
 
-# Don't change these
-# The host of the Datadog intake server to send agent data to
-default['datadog']['url'] = 'https://app.datadoghq.com'
+# Don't change these.
+# The host of the Datadog intake server to send Agent data to, only set this option
+# if you need the Agent to send data to a custom URL.
+# For Agent 6, defaults to nil because "url" overrides the site setting defined in "site".
+default['datadog']['url'] =
+  if node['datadog']['agent6'] == true
+    nil
+  else
+    'https://app.datadoghq.com'
+  end
```

Comment:
I don't think this conditional logic is needed: I don't think you need to set `default['datadog']['url']` to a non-nil value in any case given the logic you've added in the recipes.

Also, this conditional logic will be resolved early at compile time, before all the node attributes are resolved by Chef, so it will very likely give unexpected results in practice if the attribute `['datadog']['agent6']` is set in other places by users.
(in case you're wondering: the other node attributes that we do use in the present file are set by Chef before the compile phase, so they're safe to use here)

Reply:
While adding this condition I was indeed wondering if it would be safe to use `node` values here, thanks for the infos. You're right, given the others conditions added in this PR, this one is not useful.